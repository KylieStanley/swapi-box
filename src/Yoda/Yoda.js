import React from 'react'
import './Yoda.scss'


const Yoda = () => (
  <div className="yoda-container">
    <p className="choose-path">Choose a path you must...</p>
    <img src="../images/yoda.png" alt="Yoda" />
    <div className="lightsaber">
      <div className="saber" />
      <div className="handle" />
    </div>
  </div>
)

export default Yoda
