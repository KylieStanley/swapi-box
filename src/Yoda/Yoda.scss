.yoda-container {
  position: relative;
  top: 300px;
  left: 300px;
}

.choose-path {
  position: absolute;
  color: #007EB0;
  font-size: 40px;
  overflow: hidden;
  height: 100px;
  width: 500px;
  animation: type 2s steps(40, end);
  animation: glow 1s ease-in-out infinite alternate;
  animation-fill-mode: forwards;
  top: -100px;
  text-align: center;     
}

.lightsaber {
  position: absolute;
  top: -118px;
  left: -13px;
  transform: rotate(320deg)
}

.saber {
  height: 347px;
  width: 14px;
  background: rgb(135,220,90);
  background: linear-gradient(to right, rgb(135,220,90) 0%,rgb(254,254,254) 30%,rgb(254,254,254) 50%,rgb(254,254,254) 70%,rgb(135,220,90) 100%);
  filter: blur(1px);
  border-radius: 5px;       
}

.handle {
  background: grey;
  height: 64px;
  width: 16px;
  border-top: 7px solid black;
  border-bottom: 8px solid black;
  border-radius: 5px
}

@keyframes type {
  0% {
    width: 0;
  }
  99.9% {
    border-right: .15em solid #007EB0;
  }
  100% {
    border: none;
  }
}

@keyframes glow {
  from {
    text-shadow: 0 0 2px #fff, 0 0 4px #fff, 0 0 6px #007EB0, 0 0 8px #007EB0, 0 0 10px #007EB0, 0 0 12px #007EB0, 0 0 14px #007EB0;
  }
  to {
    text-shadow: 0 0 4px #fff, 0 0 6px #00b5fd, 0 0 8px #00b5fd, 0 0 10px #00b5fd, 0 0 12px #00b5fd, 0 0 14px #00b5fd, 0 0 16px #00b5fd;
  }
}
