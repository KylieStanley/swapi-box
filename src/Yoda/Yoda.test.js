import React from 'react';
import { shallow, mount } from 'enzyme'
import Yoda from './Yoda'

describe('Yoda', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Yoda />)

    expect(wrapper).toMatchSnapshot()
  })
})