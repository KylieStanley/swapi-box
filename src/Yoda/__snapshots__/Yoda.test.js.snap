// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Yoda should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Yoda />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="choose-path"
        >
          Choose a path you must...
        </p>,
        <img
          alt="Yoda"
          src="../images/yoda.png"
        />,
        <div
          className="lightsaber"
        >
          <div
            className="saber"
          />
          <div
            className="handle"
          />
        </div>,
      ],
      "className": "yoda-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Choose a path you must...",
          "className": "choose-path",
        },
        "ref": null,
        "rendered": "Choose a path you must...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Yoda",
          "src": "../images/yoda.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="saber"
            />,
            <div
              className="handle"
            />,
          ],
          "className": "lightsaber",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "saber",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "handle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="choose-path"
          >
            Choose a path you must...
          </p>,
          <img
            alt="Yoda"
            src="../images/yoda.png"
          />,
          <div
            className="lightsaber"
          >
            <div
              className="saber"
            />
            <div
              className="handle"
            />
          </div>,
        ],
        "className": "yoda-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose a path you must...",
            "className": "choose-path",
          },
          "ref": null,
          "rendered": "Choose a path you must...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Yoda",
            "src": "../images/yoda.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="saber"
              />,
              <div
                className="handle"
              />,
            ],
            "className": "lightsaber",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "saber",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "handle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
